pkg/vm/engine/memoryengine/operations.go

```diff
@@ -117,58 +124,146 @@ type CreateDatabaseReq struct {
 	CreateSql  string
 }
 
+func (m *CreateDatabaseReq) MarshalBinary() ([]byte, error) {
+	return m.Marshal()
+}
+
+func (m *CreateDatabaseReq) UnmarshalBinary(data []byte) error {
+	return m.Unmarshal(data)
+}
+
 type CreateDatabaseResp struct {
 	ID ID
 }
 
+func (m *CreateDatabaseResp) MarshalBinary() ([]byte, error) {
+	return m.Marshal()
+}
+
+func (m *CreateDatabaseResp) UnmarshalBinary(data []byte) error {
+	return m.Unmarshal(data)
+}
+
 type OpenDatabaseReq struct {
 	AccessInfo AccessInfo
 	Name       string
 }
 
+func (m *OpenDatabaseReq) MarshalBinary() ([]byte, error) {
+	return m.Marshal()
+}
+
+func (m *OpenDatabaseReq) UnmarshalBinary(data []byte) error {
+	return m.Unmarshal(data)
+}
+
 type OpenDatabaseResp struct {
 	ID        ID
 	Name      string
 	DatTyp    string
 	CreateSql string
 }
 
+func (m *OpenDatabaseResp) MarshalBinary() ([]byte, error) {
```

Comment:
What all these req/resp are not proto?   What is the benefit of hand coding these marshal/unmarsh vs protogen?

Reply:
`MarshalBinary` for `OpenDatabaseResp` is just wrapping the protogen, not hand coding. The same as others.